x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/SecurityRestFilter.java

```diff
@@ -124,21 +115,28 @@ private void doHandleRequest(RestRequest request, RestChannel channel, NodeClien
     }
 
     protected void handleException(ActionType actionType, RestRequest request, RestChannel channel, Exception e) {
-        logger.debug(() -> format("%s failed for REST request [%s]", actionType, request.uri()), e);
         threadContext.sanitizeHeaders();
-        final RestStatus restStatus = ExceptionsHelper.status(e);
+        handleException(actionType, channel, e);
+    }
+
+    public static void handleAuthenticationFailed(RestChannel channel, Exception e) {
```

Comment:
"Exports" the authn failure response building logic to be used in the channel handler.